### PR TITLE
[WIP] Bump libreoffice 6.4

### PR DESCRIPTION
**Pull request by holgersson32644**
This branch contains the bump to LO 6.4. Tthe update to version .4.1 still compiles, so please don't merge it yet.

### CODE REVIEW

File: dev-libs/libixion/libixion-9999.ebuild [hunk 1]
```diff
@@ -14,39 +14,45 @@ if [[ ${PV} == 9999 ]]; then
 	inherit git-r3 autotools
 else
 	SRC_URI="https://kohei.us/files/ixion/src/${P}.tar.xz"
-	KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~x86"
+	# depends on spdlog keywording: https://bugs.gentoo.org/708910
+	#KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~x86"
+	KEYWORDS="~amd64 ~arm64 ~ppc ~ppc64 ~x86"
+
```
Comment (a17r) [inline dev-libs/libixion/libixion-9999.ebuild]: no need for this, that's what keywording bugs are for. unnecessary empty line.

File: dev-libs/liborcus/liborcus-9999.ebuild [hunk 2]
```diff
@@ -4,51 +4,56 @@
...
 	EGIT_REPO_URI="https://gitlab.com/orcus/orcus.git"
-	inherit git-r3 autotools
+	inherit git-r3
 else
 	SRC_URI="https://kohei.us/files/orcus/src/${P}.tar.xz"
-	KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~x86"
+	# depends on spdlog keywording: https://bugs.gentoo.org/708910
+	#KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~x86"
+	KEYWORDS="~amd64 ~arm64 ~ppc ~ppc64 ~x86"
 fi
```
Comment (a17r) [inline dev-libs/liborcus/liborcus-9999.ebuild]: no need for this, that's what keywording bugs are for.

File: dev-libs/liborcus/liborcus-9999.ebuild [hunk 3]
```diff
@@ -4,51 +4,56 @@
 EAPI=7
 
 PYTHON_COMPAT=( python3_{6,7,8} )
-inherit python-single-r1
+inherit python-single-r1 autotools
```
Comment (a17r) [inline dev-libs/liborcus/liborcus-9999.ebuild]: `autotools` shouldn't be required for release package, why did you change it?

File: dev-libs/liborcus/liborcus-9999.ebuild [hunk 4]
```diff
@@ -4,51 +4,56 @@
...
 	SRC_URI="https://kohei.us/files/orcus/src/${P}.tar.xz"
-	KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~x86"
+	# depends on spdlog keywording: https://bugs.gentoo.org/708910
+	#KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~x86"
+	KEYWORDS="~amd64 ~arm64 ~ppc ~ppc64 ~x86"
 fi
 
 LICENSE="MIT"
 SLOT="0/0.15" # based on SONAME of liborcus.so
-IUSE="python +spreadsheet-model static-libs tools"
```
Comment (a17r) [inline dev-libs/liborcus/liborcus-9999.ebuild]: it is non-optional now? please describe in git commit.

File: app-office/libreoffice/libreoffice-9999.ebuild [hunk 5]
```diff
@@ -25,7 +27,10 @@ inherit autotools bash-completion-r1 check-reqs flag-o-matic java-pkg-opt-2 mult
 
 DESCRIPTION="A full office productivity suite"
 HOMEPAGE="https://www.libreoffice.org"
-SRC_URI="branding? ( https://dev.gentoo.org/~dilfridge/distfiles/${BRANDING} )"
+SRC_URI="
+	branding? ( https://dev.gentoo.org/~dilfridge/distfiles/${BRANDING} )
+	qrcode? ( https://dev-www.libreoffice.org/src/${MY_BUNDLED_QR}.tar.gz )
```
Comment (a17r) [inline app-office/libreoffice/libreoffice-9999.ebuild]: Please check `download.lst` for QRCODEGEN_TARBALL, add it to ADDONS_SRC instead.

File: app-office/libreoffice/libreoffice-9999.ebuild [hunk 6]
```diff
@@ -10,6 +10,8 @@ MY_PV="${PV/_alpha/.alpha}"
 MY_PV="${MY_PV/_beta/.beta}"
 # experimental ; release ; old
 # Usually the tarballs are moved a lot so this should make everyone happy.
+
+MY_BUNDLED_QR="QR-Code-generator-1.4.0"
```
Comment (a17r) [inline app-office/libreoffice/libreoffice-9999.ebuild]: unnecessary

File: app-office/libreoffice/libreoffice-9999.ebuild [hunk 7]
```diff
@@ -305,6 +306,10 @@ src_unpack() {
 		git-r3_fetch "${base_uri}/${PN}/help" "refs/heads/master"
 		git-r3_checkout "${base_uri}/${PN}/help" "helpcontent2" # doesn't match on help
 	fi
+	if use qrcode; then
+		# Copy the tarball unconditionally into ${S}
+		cp "${DISTDIR}/${MY_BUNDLED_QR}.tar.gz" "${S}/"
+	fi
```
Comment (a17r) [inline app-office/libreoffice/libreoffice-9999.ebuild]: unnecessary

File: dev-libs/libixion/libixion-9999.ebuild [hunk 8]
```diff
@@ -4,7 +4,7 @@
 EAPI=7
 
 PYTHON_COMPAT=( python3_{6,7,8} )
-inherit python-single-r1
+inherit python-single-r1 autotools
```
Comment (a17r) [inline dev-libs/libixion/libixion-9999.ebuild]: autotools shouldn't be required for release package, why did you change it?

File: dev-util/mdds/mdds-9999.ebuild [hunk 9]
```diff
@@ -1,9 +1,9 @@
 # Copyright 1999-2020 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=7
+EAPI="7"
```
[the remark this answers is not in the document]
Comment (holgersson32644) [reply dev-util/mdds/mdds-9999.ebuild]: Sorry, bad habit. I fixed it locally (and will bump everything at once when all things are fixed).

File: dev-util/mdds/mdds-9999.ebuild [hunk 10]
```diff
@@ -1,9 +1,9 @@
...
 
-if [[ ${PV} == 9999 ]]; then
+if [[ ${PV} == **9999 ]]; then
```
[the remark this answers is not in the document]
Comment (holgersson32644) [reply dev-util/mdds/mdds-9999.ebuild]: Fixed locally.

File: dev-util/mdds/mdds-9999.ebuild [hunk 11]
```diff
@@ -29,22 +30,24 @@ BDEPEND="
 DEPEND="dev-libs/boost:="
 RDEPEND="${DEPEND}"
 
-PATCHES=( "${FILESDIR}/${PN}-1.4.3-buildsystem.patch" )
+PATCHES=( "${FILESDIR}/${PN}-1.5.0-buildsystem.patch" )
 
-src_prepare() {
+src_prepare(){
```
[the remark this answers is not in the document]
Comment (holgersson32644) [reply dev-util/mdds/mdds-9999.ebuild]: fixed locally.

File: app-office/libreoffice/Manifest [hunk 12]
```diff
@@ -4,7 +4,14 @@ DIST 35c94d2df8893241173de1d16b6034c0-swingExSrc.zip 9796 BLAKE2B a30b0225313b23
...
+DIST libreoffice-branding-gentoo-0.8.tar.xz 151568 BLAKE2B f03c7ddeb53c5ca3fd23401679601fcf2c4037ba17be4eb7b784c7ce7ebb71a24b8ab4aac8b7da8c6b1f14dd23bc1294ba85ff4f70ad271fb4ee3c5372e10883 SHA512 785031a699b1d1895ce4b50ffc3ddf645f3a0ef9acdf37facfd18cf75db9484cb8f53a50abb63d6006ead76a80b6ff5aa99661063245ebb84bd64013d713de7f
+DIST libreoffice-help-6.3.4.2.tar.xz 13191636 BLAKE2B c1696efa0ec9fef9dbcc0c0388de8a384484af1453d200e08820ba910cdcb65638b189566a19868281fd1777213812fde06adb07c1a662302eea1d0a791ff5a0 SHA512 8c74c82c7d73adb1523300d1f7500769f5a623fdba60b440a63c122cbfeb631a6a34ecbcb115b0154a6957be1918306f7d0c1f465030f8e8cfd5c9a1acad17c9
+DIST libreoffice-help-6.4.1.2.tar.xz 88377516 BLAKE2B 04f04e8d578f07b8d011920f4b659a4b1edb67d1b7aad605b0487bd2b86181284289fe81d3ee065ec9cb418736f2700a2e78435ae169831ba97e483e0c51c09c SHA512 cfcbfe97c74ec3b91c065cc007806c55c602039b938034d8a8286b9c32301fe6c85cb46c7aafdda9ddc5286cb9a145f36e10eb70d567ca576798e88e13ee4510
+>>>>>>> app-office/libreoffice: Bump to 6.4.1.2
```
[the remark this answers is not in the document]
Comment (holgersson32644) [reply app-office/libreoffice/Manifest]: Of course. Fixed locally.

File: app-office/libreoffice/libreoffice-9999.ebuild [hunk 13]
```diff
@@ -44,6 +44,9 @@ unset DEV_URI
 # These are bundles that can't be removed for now due to huge patchsets.
 # If you want them gone, patches are welcome.
 ADDONS_SRC=(
+	# Check if we can update to the slightly less ancient version 2.3.3
+	# or drop USE="java" entirely in favor of firebird.
+	#"java? ( ${ADDONS_URI}/62c0b97e94fe47d5e50ff605d2edf37a-hsqldb-2.3.3.zip )"
```
[the remark this answers is not in the document]
Comment (holgersson32644) [reply app-office/libreoffice/libreoffice-9999.ebuild]: That tarball is within the sources upstream, and I didn't know about download.lst so far, thanks for this hint! I will drop the entire comment block and stick only with the "java? ( ... )" line from the LO 6.3 ebuild then, OK?

File: app-office/libreoffice/libreoffice-9999.ebuild [hunk 14]
```diff
@@ -63,7 +66,7 @@ unset ADDONS_SRC
 LO_EXTS="nlpsolver scripting-beanshell scripting-javascript wiki-publisher"
 
 IUSE="accessibility bluetooth +branding coinmp +cups dbus debug eds firebird
-googledrive gstreamer +gtk kde ldap +mariadb odk pdfimport postgres test
+googledrive gstreamer +gtk kde ldap +mariadb odk pdfimport postgres +qrcode test
```
[the remark this answers is not in the document]
Comment (holgersson32644) [reply app-office/libreoffice/libreoffice-9999.ebuild]: I prefer to add qrcode support with an USE flag. The cost in size is minimal, but upstream provides a flag to toggle it and I had situations within Gentoo where I didn't see use cases of people at all, so I would like to let the user decide even for a simple, non-critical feature like QR code generation. As I said, you're (AFAICT) the most active LO maintainer, so you have the last word :)

File: app-office/libreoffice/libreoffice-9999.ebuild [hunk 15]
```diff
@@ -78,8 +81,9 @@ RESTRICT="!test? ( test )"
 
 LICENSE="|| ( LGPL-3 MPL-1.1 )"
 SLOT="0"
+# waiting for ~arm, see https://bugs.gentoo.org/708910
```
[the remark this answers is not in the document]
Comment (holgersson32644) [reply app-office/libreoffice/libreoffice-9999.ebuild]: Comment is dropped locally. Will this be a keywording bug for arm for LO-6.4 then with will be blocked by keywording bugs for the deps?

File: app-text/libetonyek/libetonyek-9999.ebuild [hunk 16]
```diff
@@ -39,26 +44,22 @@ DEPEND="${RDEPEND}
 src_prepare() {
 	default
 	[[ -d m4 ]] || mkdir "m4"
-	[[ ${PV} == *9999 ]] && eautoreconf
+	[[ ${PV} == 9999 ]] && eautoreconf
```
[the remark this answers is not in the document]
Comment (holgersson32644) [reply app-text/libetonyek/libetonyek-9999.ebuild]: Mistake. Fixed locally.

File: app-text/libetonyek/libetonyek-9999.ebuild [hunk 17]
```diff
@@ -18,15 +18,20 @@ SLOT="0"
...
 
+# We must pin a single mdds version as the libetonyek build system
+# does not check against the highest available version but against
+# mdds-1.2 by default.
+MY_MDDS_PV="1.5.0"
 BDEPEND="
 	virtual/pkgconfig
 	doc? ( app-doc/doxygen )
 "
+
```
[the remark this answers is not in the document]
Comment (holgersson32644) [reply app-text/libetonyek/libetonyek-9999.ebuild]: Fixed locally.

File: app-text/libetonyek/libetonyek-9999.ebuild [hunk 18]
```diff
@@ -18,15 +18,20 @@ SLOT="0"
 IUSE="doc static-libs test"
 RESTRICT="!test? ( test )"
 
+# We must pin a single mdds version as the libetonyek build system
+# does not check against the highest available version but against
+# mdds-1.2 by default.
+MY_MDDS_PV="1.5.0"
```
[the remark this answers is not in the document]
Comment (holgersson32644) [reply app-text/libetonyek/libetonyek-9999.ebuild]: We have not, but I found it clearer to read this way and it looks to me as we can support only the newest mdds anyway. If you disagree I revert it to the if-else statement but would like to keep that comment.

File: dev-libs/libixion/libixion-9999.ebuild (hunk 1 above)
Comment (holgersson32644) [reply dev-libs/libixion/libixion-9999.ebuild]: I dropped the two comment lines locally.

File: dev-libs/liborcus/liborcus-9999.ebuild (hunk 2 above)
Comment (holgersson32644) [reply dev-libs/liborcus/liborcus-9999.ebuild]: I dropped the two comment lines locally.

File: app-office/libreoffice/libreoffice-9999.ebuild (hunk 5 above)
Comment (holgersson32644) [reply app-office/libreoffice/libreoffice-9999.ebuild]: Fixed locally.

File: app-office/libreoffice/libreoffice-9999.ebuild (hunk 6 above)
Comment (holgersson32644) [reply app-office/libreoffice/libreoffice-9999.ebuild]: Fixed locally.

File: app-office/libreoffice/libreoffice-9999.ebuild (hunk 7 above)
Comment (holgersson32644) [reply app-office/libreoffice/libreoffice-9999.ebuild]: Dropped locally.

**Comment by holgersson32644** [comment]
Hi, I fixed most, rest will follow.

Two additional things:
I changed the make call in libreoffice to emake (and dropped the "|| die") and I already built, installed and ran the LO pre-version 6.4.2.1 successfully, so both 6.4.1.2 and 6.4.2.1 work.

File: dev-libs/liborcus/liborcus-9999.ebuild (hunk 4 above)
Comment (holgersson32644) [reply dev-libs/liborcus/liborcus-9999.ebuild]: Long story short: No, it's still optional and I just introduced a bug by a typo.
I fixed the ebuild locally.

File: dev-libs/libixion/libixion-9999.ebuild (hunk 8 above)
Comment (holgersson32644) [reply dev-libs/libixion/libixion-9999.ebuild]: based on a wrong assumption. Builds without autools for releases as expected, fixed locally.

File: dev-libs/liborcus/liborcus-9999.ebuild (hunk 3 above)
Comment (holgersson32644) [reply dev-libs/liborcus/liborcus-9999.ebuild]: wrong assumption, fixed locally.

File: app-text/libetonyek/libetonyek-9999.ebuild (hunk 18 above)
[the remark this answers is not in the document]
Comment (a17r) [reply app-text/libetonyek/libetonyek-9999.ebuild]: we trigger rebuild on dev-util/mdds subslot bump so we can support as many mdds versions as we want, in theory

File: app-office/libreoffice/libreoffice-9999.ebuild (hunk 13 above)
[the remark this answers is not in the document]
Comment (a17r) [reply app-office/libreoffice/libreoffice-9999.ebuild]: thx.

File: app-office/libreoffice/libreoffice-9999.ebuild (hunk 14 above)
[the remark this answers is not in the document]
Comment (a17r) [reply app-office/libreoffice/libreoffice-9999.ebuild]: in this case, as there are no external dependencies, and the cost is insignificant, there is no gain for making it optional.

File: app-office/libreoffice/libreoffice-9999.ebuild (hunk 15 above)
[the remark this answers is not in the document]
Comment (a17r) [reply app-office/libreoffice/libreoffice-9999.ebuild]: yes.

File: app-text/libetonyek/libetonyek-9999.ebuild (hunk 18 above)
[the remark this answers is not in the document]
Comment (holgersson32644) [reply app-text/libetonyek/libetonyek-9999.ebuild]: I will revert it then.

File: app-office/libreoffice/libreoffice-9999.ebuild (hunk 14 above)
[the remark this answers is not in the document]
Comment (holgersson32644) [reply app-office/libreoffice/libreoffice-9999.ebuild]: I'll drop the USE flag then.

**Comment by a17r** [comment]
thanks!
